securedrop/requirements/update_python_dependencies

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Usage: ./update-python-dependencies
+# Run periodically to keep Python application, test, and development
+# requirements up-to-date.
+
+set -eC
+
+WORKING_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+declare -r WORKING_DIR
+declare -r VENV='review_env'
+declare -ra REQUIREMENTS_FILE_PREFIXES=('securedrop' 'test' 'develop')
+
+# This script should not be run with an active virtualenv. Calling deactivate
+# does not work reliably, so instead we quit with an error message.
+if [[ -n "${VIRTUAL_ENV}" ]]; then
+  printf '\nERROR: Please deactivate your virtualenv then re-run this script.\n'
+  exit 1
+fi
+
+# If pip and virtualenv are not available, try to install them with apt-get. If
+# apt-get is not available, quit with an error message.
+if ! command -v pip virtualenv > /dev/null; then
+  if command -v sudo apt-get > /dev/null; then
+    sudo apt-get -qq install python-virtualenv python-pip
+  else
+    printf '\nERROR: Virtualenv and pip must be installed to run this script.\n'
+    exit 1
+  fi
+fi
+
+pushd "${WORKING_DIR}"
+# Create a temporary virtualenv for the SecureDrop Python packages in our
+# requirements directory and make sure it's cleaned up on exit.
+trap 'rm -rf "${VENV}" && popd' EXIT
+
+virtualenv -p python2 "${VENV}"
+# shellcheck disable=SC1090
+source "${VENV}"/bin/activate
+pip install -U pip-tools
```

Comment:
1. This script runs faster than you can type the commands yourself. The fact we're creating and destroying a virtual environment is a small performance penalty compared to the procedural reproducibility it provides.
2. This code is highly simple and stable. The maintenance burden is pretty much negligible.
3. Scripting the dependency update process codifies the process, which is a way of both documenting the process and providing reproducible results.
4. This script will play an integral process in resolving #1617. Especially once #1617 is implemented, being able to simply run this script will be much easier than trying to remember the `pip-compile` switch which pins hashes in the generated `.txt` file, or the one which would let you pin the PyPi cert for connecting to that  (or copying commands from documentation).

> Just tested running pip-compile in a new virtualenv—it does not install dependencies into the virtualenv, which I think is further evidence that we don't need it.

This is the expected and desired behavior. Developers should maintain a persistent virtual environment that contains the `develop-requirements.txt` dependencies as described in the documentation. This script is just used (/ will just be used) to provide a safe, reproducible dependency update mechanism, and should only be run in practice by a single developer once every few weeks to months.